### PR TITLE
implemented screen annotation feature + couple of unit test cleanups

### PR DESCRIPTION
**Pull request by tobiaszuercher**
* the unit test will actually create an output that is viewable in scenarioo itself. this is useful to proove that the actual output of the cs-writer library is readable by scenarioo itself! 
* added screen annotation support.
* bugfix in build constructor

sorry next time i won't mix different things in one PR.

### CODE REVIEW

File: Scenarioo/Scenarioo/Model/Docu/Entities/ScreenRegion.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        {
+        }
+
+        public ScreenRegion(int x, int y, int width, int height)
+        {
+            X = x;
+            Y = y;
+            Height = height;
+            Width = width;
+            Width = width;
```
Comment (bruderol) [inline Scenarioo/Scenarioo/Model/Docu/Entities/ScreenRegion.cs]: @tobiaszuercher line 53 seems redundant with line 52

File: Scenarioo/Scenarioo/Model/Docu/Entities/ScreenAnnotation.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+{
+    [Serializable]
+    [XmlRoot("screenAnnotation")]
+    public class ScreenAnnotation
+    {
+        [XmlElement("region")]
+        public ScreenRegion Region { get; set; }
+
+        /// <summary>
+        /// (optional) Set the visual style of the annotation (if not set, the same style as 'neutral' will be used).
```
Comment (bruderol) [inline Scenarioo/Scenarioo/Model/Docu/Entities/ScreenAnnotation.cs]: @tobiaszuercher the style 'neutral' that is mentioned in this comment, does not exist anymore. This is an outdated comment (from me), that you probably copied from the Java library. Instead 'default' is the default. I will fix that in the Java library. Can you please fix it here?

File: Scenarioo/Scenarioo/Api/ScenarioDocuWriter.cs [hunk 3]
```diff
@@ -234,7 +234,8 @@ public void SaveScreenshot(string usecaseName, string scenarioName, Step step, b
                 this.buildName, 
                 usecaseName, 
                 scenarioName, 
-                step.StepDescription.Index);
+                step.StepDescription.Index); // TODO: filename!
```
Comment (bruderol) [inline Scenarioo/Scenarioo/Api/ScenarioDocuWriter.cs]: @tobiaszuercher Can you please open an issue for this TODO, and reference the issue number in the TODO comment.
We have the rule, that a TODO comment should always belong to an issue in the backlog for managing and tracking them. 
Thanks a lot!

File: Scenarioo/ScenariooTest/ScenariooTest/ScenarioDocuWriterTest.cs [hunk 4]
```diff
@@ -34,147 +35,153 @@
...
 
-            this.reader = new ScenarioDocuReader(this.rootDirectory);
+            reader = new ScenarioDocuReader(rootDirectory);
 
-            this.docuFiles = new ScenarioDocuFiles(this.rootDirectory);
+            docuFiles = new ScenarioDocuFiles(rootDirectory);
         }
 
-        [TestCleanup]
+        //[TestFixtureTearDown]
```
Comment (bruderol) [inline Scenarioo/ScenariooTest/ScenariooTest/ScenarioDocuWriterTest.cs]: @tobiaszuercher Can you clean this up, please? I think the [TestFixtureTeardown] is not needed anymore, instead this method should be named "cleanOutputDirectory()" which is now called on init. correct?

Another question: What is the ScenarioDocuReader? I thought the C# library does not support reading?

File: Scenarioo/Scenarioo/Model/Docu/Entities/ScreenAnnotation.cs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+        /// <summary>
+        /// (optional, but mandatory in case that clickAction is {@link ScreenAnnotationClickAction#toUrl})
+        /// The URL to browse to when the annotation is clicked on. The URL will be opened in a separate browser tab.
+        /// </summary>
+        [XmlElement("clickActionUrl")]
+        public string ClickActionUrl { get; set; }
+
+        [XmlElement("details")]
+        public Details Details { get; set; }
```
Comment (bruderol) [inline Scenarioo/Scenarioo/Model/Docu/Entities/ScreenAnnotation.cs]: @tobiaszuercher is it convenient for the client of the library, that details objects are not initialized as empty, such that he can simply add propoerties, if he likes to? I think in the Java library we initialize such objects as empty, for he can simply add properties on it.

File: Scenarioo/ScenariooTest/ScenariooTest/ScenarioDocuWriterTest.cs [hunk 6]
```diff
@@ -283,30 +356,31 @@ public void WriteTreeStructureInDetails()
...
+            Assert.IsTrue(File.Exists(docuFiles.GetScenarioFile(BranchName, BuildName, SerializationUseCase, ScenarioName)));
 
-            // THEN: the scneario file exists
-            Assert.IsTrue(File.Exists(this.docuFiles.GetScenarioFile(BranchName, BuildName, UseCaseName, ScenarioName)));
+            scenario.Status = "success";
+            writer.SaveScenario(SerializationUseCase, scenario);
         }
 
-        [TestMethod]
+        [Test] // TODO: review this async stuff, i'm not sure about that!
```
Comment (bruderol) [inline Scenarioo/ScenariooTest/ScenariooTest/ScenarioDocuWriterTest.cs]: @tobiaszuercher please introduce issues for things like this on the scenarioo-cs repo, and mention in TODO comment. Thank you!

File: Scenarioo/Scenarioo/Model/Docu/Entities/ScreenAnnotation.cs (hunk 2 above)
Comment (tobiaszuercher) [inline Scenarioo/Scenarioo/Model/Docu/Entities/ScreenAnnotation.cs]: yep! i actually wanted to ask you about that, but forgot it. perfect, i have an answer now!

File: Scenarioo/ScenariooTest/ScenariooTest/ScenarioDocuWriterTest.cs (hunk 4 above)
Comment (tobiaszuercher) [inline Scenarioo/ScenariooTest/ScenariooTest/ScenarioDocuWriterTest.cs]: yes, i'll clean it up. the tests used to delete allt he output, but you can actually point the scenarioo-ui directly to the out dir which is quite helpful.

ScenarioDocuReader:
it's possible to read, as long as there are no "Details" on the object. so most of the objects not. i don't like this at all, we already spoke about that... (json).

**Comment by bruderol** [comment]
@tobiaszuercher btw ... why is there a merge conflict? Shouldnt the two branches be up to date (except for your changes on the fork)?

**Comment by tobiaszuercher** [comment]
we changed the branch strategy slightly. thanks for all the feedback, it's integrated but we won't merge that directly to master. itts already on the release branch which created a release candidate package.

**Pull request closed without merging**